### DISCUSSION TITLE
Spacecraft Systems Interfacing

### DISCUSSION THREAD

**Discussion by geoffreygarrett** [tudat-team/tudat#97] [Ideas]
As per our discussion during the monthly tudat developer meeting, we are moving towards a development run which would allow users to:

> interface the inter-dependencies that spacecraft systems have with the propagation of spacecraft dynamics, as per the dependencies resulting from/on guidance, perturbations, and termination conditions (i.e. structural failure).

Those who attended during the meeting (aside from myself), I will mention here, to get this discussion thread started:
- @transferorbit
- @DominicDirkx
- @jgte
- @gaffarelj
- @MarieFayolle 
- @MiguelAvillez

Perhaps a guiding mission statement similar to the above!

**Comment by DominicDirkx** [comment]
As mentioned in this thread:

https://tudat.slack.com/archives/C02140PKDEJ/p1653066929107949

I want to modify how the body-fixed thrust direction and magnitude is stored in Tudat, which is related to the discussion here.

I see (at least) two types of system models:
* Model for the exterior shape of a body, and its relation to radiation pressure forces (including thermal), and aerodynamic forces. These models are *not* spacecraft-specific, and would also be relevant for e.g. small solar system bodies. A panelled model, like the one @jgte discussed at the previous developer meeting, would seem to be a good option for this. Each panel could be endowed with any number of properties, which would allow calculations of various types of forces/interactions
* A model for a specific system which can be, but need not be, on the outside of a body. Examples:
   * Engine models (thrust direction, thrust magnitude, mass flow)
   * Tank models (mass rate, which influences e.g. inertia tensor)
   * Optical instruments (boresight, FoV)
   * ...
For the second category, each of these could (but need not) have a 
* Mass
* Location (w.r.t. CoM)
* ...
What are your thoughts on the above?

**Comment by gaffarelj** [comment]
This list would really take potential applications of Tudat to a new level I think (or at least open up new complex research possibilities).
One additional point that I think would help in some research/use cases is to add an electrical model:

- Solar power input (can be linked to the new paneled model).
- Stored energy in batteries.
- User-defined energy draw from the battery and/or solar panels.
- System "activation laws" (better name pending) to define when to use a given system, based on any number of dependent variables (existing ones like altitude, velocity, Kepler element, but also including solar panel power input and battery capacity). These systems that are activated could be any of the ones @DominicDirkx listed above that are power-dependent (optical payload, electrical engine)
